msal/wam.py

```diff
@@ -0,0 +1,159 @@
+    params.set_requested_scopes(scope or "https://graph.microsoft.com/.default")
+    params.set_redirect_uri(
+        "https://login.microsoftonline.com/common/oauth2/nativeclient")
+    callback_data = _CallbackData()
+    pymsalruntime.signin_interactively(
+        window or win32gui.GetDesktopWindow(),  # TODO: Remove win32gui
+        params,
+        "correlation", # TODO
+        login_hint or "",  # Account hint
+        lambda result, callback_data=callback_data: callback_data.complete(result))
+    callback_data.signal.wait()
+    return _convert_result(callback_data.auth_result)
+
+
+def _acquire_token_silently(authority, client_id, account, scope):
+    params = pymsalruntime.MSALRuntimeAuthParameters(client_id, authority)
+    params.set_requested_scopes(scope)
+    callback_data = _CallbackData()
+    pymsalruntime.signin_silently(
+        params,
+        "correlation", # TODO
+        lambda result, callback_data=callback_data: callback_data.complete(result))
+    callback_data.signal.wait()
+    return _convert_result(callback_data.auth_result)
+
+
+def _acquire_token_interactively(
+        authority,
+        client_id,
+        account,
+        scopes,
+        prompt=None,  # TODO: Perhaps WAM would not accept this?
+        login_hint=None,  # type: Optional[str]
+        domain_hint=None,  # TODO: Perhaps WAM would not accept this?
+        claims_challenge=None,
+        timeout=None,  # TODO
+        extra_scopes_to_consent=None,  # TODO: Perhaps WAM would not accept this?
+        max_age=None,  # TODO: Perhaps WAM would not accept this?
+        **kwargs):
+    params = pymsalruntime.MSALRuntimeAuthParameters(client_id, authority)
+    params.set_requested_scopes(" ".join(scopes))
+    if login_hint:
+        params.set_login_hint(login_hint)
+    if claims_challenge:
+        params.set_claims(claims_challenge)
+    # TODO: Wire up other input parameters too
+    callback_data = _CallbackData()
+    pymsalruntime.acquire_token_interactively(
+        window,  # TODO
+        params,
+        "correlation", # TODO
+        account,
+        lambda result, callback_data=callback_data: callback_data.complete(result))
+    callback_data.signal.wait()
+    return callback_data.auth_result
+
+
+def acquire_token_interactive(
+        authority,  # type: str
+        client_id,  # type: str
+        scopes,  # type: list[str]
+        login_hint=None,
+        **kwargs):
+    """MSAL Python's acquire_token_interactive() will call this"""
+    return _signin_interactively(
+        authority,
+        client_id,
+        " ".join(scopes),
+        login_hint=login_hint)
+
+
+def acquire_token_silent(
+        authority,  # type: str
+        client_id,  # type: str
+        scopes,  # type: list[str]
+        account=None,  # TBD
+        ):
+    scope = " ".join(scopes)
+    if account:
+        wam_account = _read_account_by_id(account["some_sort_of_id"])  # TODO
```

Comment:
>some_sort_of_id [](http://example.com/codeflow?start=51&length=15)

This is the localAccountId which is returned from the mid-tier get_accountID API.